### ISSUE TITLE
lib's problem: the host header of websocket client is lack of port

### ISSUE THREAD

**Issue by x1aohe1** [uNetworking/uWebSockets#623]
It is not a issue if you write your own websocket server and useless of host header, but when i 
use autobahn as the server, it need host header like this : "ws://hostname:port" ; then this problem occur when i run the client which i wrote: "FAILURE: 0x8 should not emit error!" ( 0x8 is a variables) .

  So, my only way is debug the lib, i print some info in HTTPSocket.cpp's onData() function, 
(i didn't wrote onMessage(), so i don't know if the error message will return to onMessage(), you can try it) . result is the server with autobahn return the message like this"HTTP/1.1 400 missing port in HTTP Host header 'host' 
and server runs on non-standard port xxxx" , then i use another client i wrote in js, it's header is create by 
the js or something i don't know, it's host header is like what i said above. Then , i overwrite the header 
which is in Hub.cpp 's 141 line like "httpSocket->httpBuffer ......"Host: " + hostname + ":" + portStr + "\r\n"",
    after all, compile the lib and run the .out , BINGO!! it work!!

**Comment by alexhultman** [comment]
TLDR; The client doesn't pass Autobahn?

**Comment by x1aohe1** [comment]
TLDR?

**Comment by x1aohe1** [comment]
TLDR mean？
yes, because server use autobahn need the client request's host header to have PORT  or fail to connect it

**Comment by aldanor** [comment]
I think @x1aohe1 wants to connect to a `ws://hostname:port` web socket server which runs on a non-standard (non-80 / non-44) port, in which case the upgrade request should encode this in the "Host" header, like `Host: hostname:port`.
